scripts/xtensa-build-all.sh

```diff
@@ -113,171 +89,142 @@ set -e
 OLDPATH=$PATH
 WORKDIR="$pwd"
 
-# build platform
-for j in ${PLATFORMS[@]}
+# build platforms
+for platform in "${PLATFORMS[@]}"
 do
 	HAVE_ROM='no'
 	DEFCONFIG_PATCH=''
-	if [ $j == "byt" ]
-	then
-		PLATFORM="baytrail"
-		ARCH="xtensa"
-		XTENSA_CORE="Intel_HiFiEP"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
-		HOST="xtensa-byt-elf"
-		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
-	fi
-	if [ $j == "cht" ]
-	then
-		PLATFORM="cherrytrail"
-		ARCH="xtensa"
-		XTENSA_CORE="CHT_audio_hifiep"
-		ROOT="$pwd/../xtensa-root/xtensa-byt-elf"
-		HOST="xtensa-byt-elf"
-		XTENSA_TOOLS_VERSION="RD-2012.5-linux"
-	fi
-	if [ $j == "bdw" ]
-	then
-		PLATFORM="broadwell"
-		ARCH="xtensa"
-		XTENSA_CORE="LX4_langwell_audio_17_8"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
-		HOST="xtensa-hsw-elf"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-	fi
-	if [ $j == "hsw" ]
-	then
-		PLATFORM="haswell"
-		ARCH="xtensa"
-		XTENSA_CORE="LX4_langwell_audio_17_8"
-		ROOT="$pwd/../xtensa-root/xtensa-hsw-elf"
-		HOST="xtensa-hsw-elf"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-	fi
-	if [ $j == "apl" ]
-	then
-		PLATFORM="apollolake"
-		ARCH="xtensa-smp"
-		XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
-
-		# test APL compiler aliases and ignore set -e here
-		type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
-		if [ $? == 0 ]
-		then
-			HOST="xtensa-bxt-elf"
-		else
-			HOST="xtensa-apl-elf"
-		fi
 
-		ROOT="$pwd/../xtensa-root/$HOST"
-		XTENSA_TOOLS_VERSION="RG-2017.8-linux"
-		HAVE_ROM='yes'
-	fi
-	if [ $j == "skl" ]
-	then
-		PLATFORM="skylake"
-		ARCH="xtensa"
-		XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
-
-		# test APL compiler aliases and ignore set -e here
-		type xtensa-bxt-elf-gcc > /dev/null 2>&1 && true
-		if [ $? == 0 ]
-		then
-			HOST="xtensa-bxt-elf"
-		else
-			HOST="xtensa-apl-elf"
-		fi
+	case $platform in
+		byt)
+			PLATFORM="baytrail"
+			ARCH="xtensa"
+			XTENSA_CORE="Intel_HiFiEP"
+			HOST="xtensa-byt-elf"
+			XTENSA_TOOLS_VERSION="RD-2012.5-linux"
+			;;
+		cht)
+			PLATFORM="cherrytrail"
+			ARCH="xtensa"
+			XTENSA_CORE="CHT_audio_hifiep"
+			HOST="xtensa-byt-elf"
+			XTENSA_TOOLS_VERSION="RD-2012.5-linux"
+			;;
+		bdw)
+			PLATFORM="broadwell"
+			ARCH="xtensa"
+			XTENSA_CORE="LX4_langwell_audio_17_8"
+			HOST="xtensa-hsw-elf"
+			XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+			;;
+		hsw)
+			PLATFORM="haswell"
+			ARCH="xtensa"
+			XTENSA_CORE="LX4_langwell_audio_17_8"
+			HOST="xtensa-hsw-elf"
+			XTENSA_TOOLS_VERSION="RG-2017.8-linux"
+			;;
+		apl)
+			PLATFORM="apollolake"
+			ARCH="xtensa-smp"
+			XTENSA_CORE="X4H3I16w2D48w3a_2017_8"
+
+			# test APL compiler aliases and ignore set -e here
+			if ! type xtensa-bxt-elf-gcc; then
```

Comment:
I think you meant: `if type xtensa-bxt-elf-gcc; then`.
This explains the Travis failure.

Reply:
Either way `HOST` is never set to `xtensa-bxt-elf-gcc` which is what is being set as the compiler. Currently unable to get the docker image because of network issues so cannot reproduce locally. Anyways, I will try removing the `!` and see if it works
```
  The CMAKE_C_COMPILER:

    xtensa-bxt-elf-gcc

  is not a full path and was not found in the PATH.

```

Reply:
huh, this does fix it. But doesn't work well with what I said earlier. Maybe the `CMAKE_C_COMPILER` is not derived from `TOOLCHAIN` which is what is using the `HOST` value?

Reply:
`bxt` and `apl` were technically different but have been used somewhat interchangeably in some contexts. If you look at commit c7edd736d9c6 you can see this logic is only backward compatibility to support who built a toolchain with the older `bxt` name. All documentation now refers to `apl`